examples/pytorch/graphsaint/sampler.py

```diff
@@ -10,23 +10,74 @@
 
 
 # The base class of sampler
-# (TODO): online sampling
-class SAINTSampler(object):
-    def __init__(self, dn, g, train_nid, node_budget, num_repeat=50):
-        """
-        :param dn: name of dataset
-        :param g: full graph
-        :param train_nid: ids of training nodes
-        :param node_budget: expected number of sampled nodes
-        :param num_repeat: number of times of repeating sampling one node
-        """
-        self.g = g
+class SAINTSampler:
+    """
+    Description
+    -----------
+    SAINTSampler implements the sampler described in GraphSAINT. This sampler implements offline sampling in
+    pre-sampling phase as well as fully offline sampling, fully online sampling in training phase.
+    Users can conveniently set param 'online' of the sampler to choose different modes.
+
+    Parameters
+    ----------
+    dn : str
+        name of dataset.
+    g : DGLGraph
+        the full graph.
+    train_nid : list
+        ids of training nodes.
+    num_workers_sampler : int
+        number of processes to sample subgraphs in pre-sampling procedure using torch.dataloader.
+    train : bool
+        a flag specifying if the sampler is utilized in training procedure or not. Note that the sampling methods are
+        different in pre-sampling phase and training phase respectively. We have to assign this flag as False before
+        pre-sampling phase and True after pre-sampling before training.
+    num_subg_norm : int, optional
+        the max number of subgraphs sampled in pre-sampling phase for computing normalization coefficients in the beginning.
+        Actually this param is used as `__len__` of sampler in pre-sampling phase.
+        Please make sure that num_subg_norm is greater than batch_size_norm so that we can sample enough subgraphs.
+        Defaults: 10000
+    batch_size_norm : int, optional
+        the number of subgraphs sampled by each process concurrently in pre-sampling phase.
+        Defaults: 200
+    online : bool, optional
+        This param is valid only when `train` is True.
+        If `True`, we employ fully online sampling in training phase. Otherwise employing fully offline sampling.
```

Comment:
I wonder what will be a non-fully online/offline sampling.

Reply:
Ahhh I'll modify the annotations later lol Actually there is previously a sampling method combining the offline sampling and online sampling so this wrong annotation appears here.